[bddisasm] new port

ianichitei opened the pull request:
- #### What does your PR fix?  

Create a port of https://github.com/bitdefender/bddisasm as discussed in microsoft/vcpkg#16982.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  

- Windows

  - x64-windows
  - x86-windows
  - x64-windows-static
  - x86-windows-static

- Linux

  - x64-linux

I'm not entirely sure I did the right thing here. The library can be built only as a static library. Should that be expressed in some way in the port file? As it stands, `x64-windows` and `x64-windows-static` are the same thing. This feels wrong, but I can't figure out what's the right way to do things in this case. Should the install fail if `VCPKG_LIBRARY_LINKAGE` is `dynamic`? The [docs](https://vcpkg.readthedocs.io/en/latest/users/triplets/) say that "libraries can ignore this setting if they do not support the preferred linkage type", so I'm thinking that probably not.

For the same reason, I'm not entirely sure what should I use for the `supports` field of the manifest file.

I haven't update `ci.baseline.txt`. I'm not entirely sure how that works (I'm fairly new to vcpkg, but if I'm pointed in the right direction I'll probably do the right thing).

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  

Yes.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

ianichitei committed:
Add bddisasm port

Currently fetching bddisasm from my fork of the project

ianichitei committed:
bddisasm: Build with BDD_INCLUDE_TOOL=OFF

ianichitei committed:
bddisasm: Use PREFER_NINJA

ianichitei committed:
bddisasm: Update REF

ianichitei committed:
bddisasm: Don't use vcpkg_install_msbuild

ianichitei committed:
bddisasm: Switch to release 1.32.1 from the bitdefender repo

ianichitei committed:
Merge branch 'microsoft:master' into master

ianichitei committed:
Add the result of ./vcpkg x-add-version --all

ianichitei committed:
Replace the control file with a manifest

Converted with ./vcpkg format-manifest

ianichitei committed:
Add more details to bddisasm/vcpkg.json

Added port-version, license, homepage, documentation, and port mantainers

NancyLi1013 commented:
Please ping me if this is ready for review.

ianichitei committed:
Fix bddisasm/vcpkg.json

ianichitei committed:
Fix the version file

ianichitei committed:
List arm as not supported

ianichitei committed:
Update version file

ianichitei commented:
> Please ping me if this is ready for review.

All the checks have passed now.

NancyLi1013 commented on ports/bddisasm/LICENSE:
```diff
@@ -0,0 +1,201 @@
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
```
I noticed that there is LICENSE file in source codes. So it's unnecessary to add it here.

https://github.com/bitdefender/bddisasm/blob/master/LICENSE

NancyLi1013 commented on ports/bddisasm/portfile.cmake:
```diff
@@ -0,0 +1,29 @@
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
```
```suggestion
vcpkg_cmake_configure(
```
We prefer to use new function instead of the deprecated one for new ports. 
https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_configure_cmake.md

ianichitei replied:
I feel like I wasted your time by using the old functions. I did a first draft of this port sometimes in 2020, but haven't made it public until recently. `vcpkg_cmake_configure` wasn't a thing back than (at least I don't remember). I should have checked for updates before opening the PR.

NancyLi1013 replied:
For the deprecated functions, this is due to that we have no document to declare we should use new functions for new ports. So this is not wasting time. We need to pick this point up when review PRs. As you can see, there are still many ports using the deprecated functions. Currently, we only suggest to use the new functions for new ports. Other cases are optional.

NancyLi1013 commented on ports/bddisasm/portfile.cmake:
```diff
@@ -0,0 +1,29 @@
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    DISABLE_PARALLEL_CONFIGURE
+    PREFER_NINJA
```
```suggestion
```
This argument is not needed if we use `vcpkg_cmake_configure()`. Since ninja is the default build system. Please see 
https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/ports/vcpkg-cmake/vcpkg_cmake_configure.md

NancyLi1013 commented on ports/bddisasm/portfile.cmake:
```diff
@@ -0,0 +1,29 @@
+    SOURCE_PATH ${SOURCE_PATH}
+    DISABLE_PARALLEL_CONFIGURE
+    PREFER_NINJA
+    OPTIONS -DBDD_INCLUDE_TOOL=OFF
```
```suggestion
    OPTIONS 
        -DBDD_INCLUDE_TOOL=OFF
```

NancyLi1013 commented on ports/bddisasm/portfile.cmake:
```diff
@@ -0,0 +1,29 @@
+    OPTIONS -DBDD_INCLUDE_TOOL=OFF
+)
+
+vcpkg_install_cmake()
```
```suggestion
vcpkg_cmake_install()
```
Please use new function instead of the deprecated one.

NancyLi1013 commented on ports/bddisasm/portfile.cmake:
```diff
@@ -0,0 +1,29 @@
+vcpkg_install_cmake()
+
+file(INSTALL
+    ${CURRENT_PORT_DIR}/LICENSE
```
```suggestion
    ${SOURCE_PATH}/LICENSE
```

NancyLi1013 commented on ports/bddisasm/portfile.cmake:
```diff
@@ -0,0 +1,29 @@
+    RENAME copyright
+)
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/bddisasm TARGET_PATH share/bddisasm)
```
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/bddisasm)
```
Please use the new function here. `TARGET_PATH `defaults to `share/${PORT}`. So it's unnecessary to add it here.

NancyLi1013 commented on ports/bddisasm/vcpkg.json:
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "bddisasm",
+  "version-string": "1.31.8",
```
```suggestion
  "version": "1.32.1",
```
The latest release should be 1.32.1. For field name, please refer to https://github.com/microsoft/vcpkg/blob/master/docs/users/versioning.md

NancyLi1013 commented on versions/b-/bddisasm.json:
```diff
@@ -0,0 +1,9 @@
+  "versions": [
+    {
+      "git-tree": "ec47a13b95291dd3ff61b7a9b9d5f95b3e011d3a",
+      "version-string": "1.31.8",
```
```suggestion
      "version": "1.32.1",
```

NancyLi1013 commented on ports/bddisasm/vcpkg.json:
```diff
@@ -0,0 +1,10 @@
+  "documentation": "https://bddisasm.readthedocs.io/en/latest/",
+  "license": "Apache-2.0",
+  "supports": "!arm"
+}
```
If we use new functions, we should also add vcpkg-cmake, vckg-cmake-config to the dependencies:

   ```
"dependencies": [
    {
      "name": "vcpkg-cmake",
      "host": true
    },
    {
      "name": "vcpkg-cmake-config",
      "host": true
    }
```

NancyLi1013 commented:
Please convert this PR to open instead of draft if it is ready for review. @ianichitei

NancyLi1013 commented on ports/bddisasm/portfile.cmake:
```diff
@@ -0,0 +1,29 @@
+vcpkg_from_github(
```
```suggestion
vcpkg_fail_port_install(ON_ARCH "arm")

vcpkg_from_github(
```
Seems it doesn't support arm, it would be better to add the unsupported message here.

ianichitei committed:
Fail install on arm

ianichitei committed:
Use the new vcpkg_cmake_* functions

ianichitei committed:
Remove the LICENSE file

ianichitei committed:
Fix version

ianichitei committed:
Update version file

NancyLi1013 commented:
LGTM now, thanks for your adding this new port. @ianichitei

ianichitei commented:
> LGTM now, thanks for your adding this new port. @ianichitei

Thanks for helping me :)

strega-nil committed:
build only static library

bddisasm only supports being built as a static library, so say that in the portfile
